src/Chapter2.hs

```diff
@@ -621,7 +630,15 @@ Write a function that takes elements of a list only in even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+
+takeEven :: [a] -> [a]
+takeEven l = go l []
+  where
+    go :: [a] -> [a] -> [a]
+    go [] ls = ls
+    go [y] ls = ls ++ [y]
+    go (y : _ : xs) ls = go xs ls ++ [y]
```

Comment:
Nice idea on the helper functions :+1:

Alternatively, you could use pattern matching in `takeEven` instead. As you need to take every second element, it is possible to represent with a few pattern matching clauses 🙂
This will also be more efficient, as you won't need to add to the end of the list, which is very slow.